aydin/nn/models/torch/torch_jinet.py

```diff
@@ -0,0 +1,291 @@
+from collections import OrderedDict
+from itertools import chain
+
+import torch
+from torch import nn
+from torch.optim.lr_scheduler import ReduceLROnPlateau
+from torch.utils.tensorboard import SummaryWriter
+
+from aydin.nn.layers.dilated_conv import DilatedConv
+from aydin.nn.pytorch.optimizers.esadam import ESAdam
+from aydin.util.log.log import lprint
+
+
+class JINetModel(nn.Module):
+    def __init__(
+        self,
+        spacetime_ndim,
+        nb_in_channels: int = 1,
+        nb_out_channels: int = 1,
+        kernel_sizes=None,
+        num_features=None,
+        nb_dense_layers: int = 3,
+        nb_channels: int = None,
+        final_relu: bool = False,
+        degressive_residuals: bool = False,  # TODO: check what happens when this is True
+    ):
+        super(JINetModel, self).__init__()
+
+        self.spacetime_ndim = spacetime_ndim
+        self.nb_in_channels = nb_in_channels
+        self.nb_out_channels = nb_out_channels
+        self._kernel_sizes = kernel_sizes
+        self._num_features = num_features
+        self.nb_dense_layers = nb_dense_layers
+        self.nb_channels = nb_channels
+        self.final_relu = final_relu
+        self.degressive_residuals = degressive_residuals
+
+        if len(self.kernel_sizes) != len(self.num_features):
+            raise ValueError("Number of kernel sizes and features does not match.")
+
+        self.dilated_conv_functions = []
```

Comment:
It might be safer to use `nn.ModuleList`? You can keep it if you didn't get error with `model.to('cuda')`.

Reply:
definitely, good catch!